### DISCUSSION TITLE
Notifications for PRs in fine-grained control, as in CODEOWNERS

### DISCUSSION THREAD

**Discussion by hamogu** [community/community#16370] [Pull Requests]
I'm looking for a way to have notifications for any PR that touches a specific sub-package of a large repository. 
Currently, I can "watch" an entire repro, which is a lot of traffic for large repositories, or "subscribe" to individual issues or PRs - but that requires me to monitor the issue and PR list manually.

I have tow use cases for this functionality:
- I'm a sub-package maintainer for a larger package but I'm drowning in emails if I watch the entire package. This case is somewhat related to the use case for CODEOWNERS, but I don't want to be requested for review automatically. I'm a volunteer and I decide myself which PRs I take on; often others, who are not the designated maintainer will take on the initial review of a PR, but won't do that if I'm already listed.
- I'm interested in a sub-package that I'm not (or only loosely) affiliated with. In my case, I maintain a package that uses just one sub-package of a large upstream package. Again, if I watch that entire upstream package, I'll drown in emails, but I do want to know about any change in the sub-package I care about, so that I can test if that breaks my downstream package and offer comments and suggestions.

CODEOWNERS already has a mechanism to select specific directories (typically sub-packages) and be requested for review for all PRs that touch that sub-package. So, one way to do what I want would be to have a CODENOTIFY file with the same syntax as CODEOWNERS, but people listed there would only be @-mentioned and not requested for review. (Alternatively, CODEOWNERS could get a syntax to select "notify only" instead of "request for review.)

For use case 2, it would be better to have that control through the github webpage (like clicking the "watch" and "subscribe" buttons), but I realize that it's not easy to have a clean API for that. A CODENOTIFY in the upstream package would do for that, I think. If I'm that interested, it's likely that I already have some relation with the upstream package and they would allows a PR to their CODENOTIFY file from me.

**Comment by ag-eitilt** [comment]
I'd like to second this request.  My team and I are responsible for (among other things) a tool which takes as input a large and non-closed selection of files in the primary repository, and would like to be notified of changes to these files to keep track of how the tool is being used, and to be able to suggest better ways of doing so if necessary.  However, we don't consider ourselves *responsible* for those input files once the code's merged, and especially in the sense of being the ones to provide ongoing maintenance/be contacted with questions on their contents.  We're currently able to set up a team "ownership" which people just know is not a full ownership, but it would be ideal if there were a lesser level where we were able to assign ourselves to be notified on change, but not be listed as having the ultimate responsibility for the code at rest.  In our use case, it would likely be more similar to a Gerrit +1/+2 system (we're fine being requested as reviewers), but a CODENOTIFY would certainly do the job very nicely as well.
